pet_code/scripts/tb_pet_validation.py

```diff
@@ -0,0 +1,271 @@
+#!/usr/bin/env python3
+
+"""Calculate and save at slab level: Energy Res., Centroid, CTR, min. and max. cog in monolithic direction
+
+Usage: tb_pet_validation.py (--conf CONFFILE) INFILE...
+
+Arguments:
+    INFILE  File name to be analysed
+
+Required:
+    --conf=CONFFILE  Configuration file for run.
+"""
+
+import os
+import configparser
+
+from docopt    import docopt
+from itertools import repeat
+
+import matplotlib.pyplot as plt
+import numpy as np
+import math
+from scipy.signal import find_peaks
+from colorama import Fore, Style
+
+
+from pet_code.src.io    import read_petsys_filebyfile
+from pet_code.src.io    import read_ymlmapping
+from pet_code.src.plots import mm_energy_spectra
+from pet_code.src.util  import calibrate_energies
+from pet_code.src.util  import centroid_calculation
+from pet_code.src.util  import filter_event_by_impacts
+from pet_code.src.util  import filter_impacts_one_minimod
+from pet_code.src.util  import mm_energy_centroids
+from pet_code.src.util  import select_module
+from pet_code.src.util  import select_array_range
+from pet_code.src.fits  import fit_gaussian
+
+
+import time
+
+
+def sm_slab_specs(sm):
+    """Takes the sm event by event information and retrieve slab performance: compresion,
+    energy resolution (ER) and centroid of the energy distribution per miniModule.
+    Input: 
+    sm: supermodule in dictionary form.
+    Returns: 
+    slab_params: dictionary with specs per slab: [compresion in monolithic (ymin, ymax), compresion
+    in pixelated (xmin, xmax), mu, ER].
+    roi_mm_dict: dictionary with the ROI per slab to be plotted in the floodmap.
+    """
+
+    slab_params    = {}
+    roi_mm_dict    = {}
+    mm_compression = {}
+    for mm in sm:        
```

Comment:
Usaría `for mm, mm_vals in sm.items()` para tener los valores relevantes siempre.